### PR TITLE
feat: list_debuggees supports active debuggees

### PR DESCRIPTION
**Pull request by jasonborg**
In support of https://github.com/GoogleCloudPlatform/snapshot-debugger/issues/76 the `list_debuggees` command now:
- Shows only active debuggees by default
- Supports a `--include-inactive` flag
- Sorts the output with the most recently active debuggees being shown
  first.

### CODE REVIEW

File: snapshot_dbg_cli/snapshot_debugger_rtdb_service.py [hunk 1]
```diff
@@ -50,8 +51,18 @@ def get_schema_version(self):
...
     return self.rest_service.set(self.schema.get_path_schema_version(), version)
 
-  def get_debuggees(self):
-    return self.rest_service.get(self.schema.get_path_debuggees())
+  def get_debuggees(self, current_time_unix_msec):
+    debuggees = self.rest_service.get(self.schema.get_path_debuggees()) or {}
+
+    # The result will be a dictionary, convert it to an array, while also
+    # filtering out any invalid entries, normalize_debuggee returns None for
+    # invalid debuggee.
```
Comment (mctavish) [inline snapshot_dbg_cli/snapshot_debugger_rtdb_service.py:59]: There's at least one too many `,`s in here.  I think that it parses better if "normalize_debuggee ..." is the start of a new sentence.

File: snapshot_dbg_cli_tests/test_debuggee_utils.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+
+  def test_set_converted_timestamps(self):
+    """Verifies the set_converted_timestamps() works as expected.
+
+    If the human readable time is not in the breakpoint, but the unix msec
+    version is, the human readable version should be populated, otherwise no
+    changes are done.
+    """
+
+    # Note, these tests don't need fully flesched out breakpoints, the function
```
Comment (mctavish) [inline snapshot_dbg_cli_tests/test_debuggee_utils.py:37]: s/flesched/fleshed

File: snapshot_dbg_cli_tests/test_debuggee_utils.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+            {
+                'registrationTimeUnixMsec': 1649962215000,
+                'registrationTime': '2022-04-14T18:50:15.000000Z'
+            },
+        ),
+        (
+            'registrationTime not overrwritten',
+            {
+                'registrationTimeUnixMsec': 1649962215000,
+                'registrationTime': 'exists not overwritten'
```
Comment (mctavish) [inline snapshot_dbg_cli_tests/test_debuggee_utils.py:56]: In what cases will the human readable form be set before conversion?

File: snapshot_dbg_cli/debuggee_utils.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    # time was not actually known.
+    debuggee['registrationTimeUnixMsec'] = 0
+
+  if 'lastUpdateTimeUnixMsec' not in debuggee:
+    # Debuggees created by older agents won't have this field present,
+    # initialize it to 0 so it's set to something, but it's clear the
+    # time was not actually known.
+    debuggee['lastUpdateTimeUnixMsec'] = 0
+
+  set_converted_timestamps(debuggee)
```
Comment (mctavish) [inline snapshot_dbg_cli/debuggee_utils.py:99]: Looks to me like we're going to slap some January 1st, 1970 strings into the debuggees.  I'd argue that it's better to set the string to "not set" or blank.

File: snapshot_dbg_cli/COMMAND_REFERENCE.md [hunk 5]
```diff
@@ -32,16 +32,18 @@ snapshot-cdbg-cli list_debuggees
 
 
 Usage: `__main__.py list_debuggees [-h] [--database-url DATABASE_URL] [--format
-FORMAT] [--debug]`
+FORMAT] [--debug] [--include-inactive]`
 
 Used to display a list of the debug targets (debuggees) registered with the
-Snapshot Debugger.
+Snapshot Debugger. By default all active debuggees are returned. To also obtain
```
[the remark this answers is not in the document]
Comment (jasonborg) [reply snapshot_dbg_cli/COMMAND_REFERENCE.md:38]: Good call.

File: snapshot_dbg_cli/snapshot_debugger_rtdb_service.py (hunk 1 above)
Comment (jasonborg) [reply snapshot_dbg_cli/snapshot_debugger_rtdb_service.py:59]: Good call.

File: snapshot_dbg_cli_tests/test_debuggee_utils.py (hunk 3 above)
Comment (jasonborg) [reply snapshot_dbg_cli_tests/test_debuggee_utils.py:56]: They won't be in practice, but the conversion function was written defensively to not overwrite it if it is set.

File: snapshot_dbg_cli/debuggee_utils.py (hunk 4 above)
Comment (jasonborg) [reply snapshot_dbg_cli/debuggee_utils.py:99]: Yeah, makes sense, I figured seeing 1970... would be an indication of that, and it keeps the same format, but nothing depends on that format, it purely for human consumption, so I'll update it.

**Pull request merged**